src/set/tests.rs

```diff
@@ -583,6 +583,264 @@ fn iter_default() {
+    assert_eq!(index_set.len(), 0);
+
+    let result = index_set.take(&20);
+    assert_eq!(result, None);
+}
+
+#[test]
+fn swap_take() {
+    let mut index_set: IndexSet<i32> = IndexSet::new();
+    index_set.insert(10);
+    index_set.insert(20);
+    index_set.insert(30);
+    index_set.insert(40);
+    assert_eq!(index_set.len(), 4);
+
+    let result = index_set.swap_take(&20);
+    assert_eq!(result, Some(20));
+    assert_eq!(index_set.len(), 3);
+    assert_eq!(index_set.as_slice(), &[10, 40, 30]);
+
+    let result = index_set.swap_take(&50);
+    assert_eq!(result, None);
+}
+
+#[test]
+fn sort_unstable() {
+    let mut index_set: IndexSet<i32> = IndexSet::new();
+    index_set.insert(30);
+    index_set.insert(20);
+    index_set.insert(10);
+
+    index_set.sort_unstable();
+    assert_eq!(index_set.as_slice(), &[10, 20, 30]);
+}
+
+#[test]
+fn try_reserve_exact() {
+    let mut index_set: IndexSet<i32> = IndexSet::new();
+    index_set.insert(10);
+    index_set.insert(20);
+    index_set.insert(30);
+    assert_eq!(index_set.capacity(), 3);
+
+    index_set.try_reserve_exact(2).unwrap();
+
+    assert_eq!(index_set.capacity(), 5);
+}
+
+#[test]
+fn shift_remove_full() {
+    let mut set: IndexSet<i32> = IndexSet::new();
+    set.insert(10);
+    set.insert(20);
+    set.insert(30);
+    set.insert(40);
+    set.insert(50);
+
+    let result = set.shift_remove_full(&20);
+    assert_eq!(result, Some((1, 20)));
+    assert_eq!(set.len(), 4);
+
+    set.swap_remove(&30);
+
+    let result = set.shift_remove_full(&50);
+    assert_eq!(result, Some((1, 50)));
+    assert_eq!(set.len(), 2);
+
+    let result = set.shift_remove_full(&60);
+    assert_eq!(result, None);
+    assert_eq!(set.len(), 2);
+}
+
+#[test]
+fn shift_remove_index() {
+    let mut set: IndexSet<i32> = IndexSet::new();
+    set.insert(10);
+    set.insert(20);
+    set.insert(30);
+    set.insert(40);
+    set.insert(50);
+
+    let result = set.shift_remove_index(1);
+    assert_eq!(result, Some(20));
+    assert_eq!(set.len(), 4);
+
+    set.swap_remove(&30);
+
+    let result = set.shift_remove_index(1);
+    assert_eq!(result, Some(50));
+    assert_eq!(set.len(), 2);
+
+    let result = set.shift_remove_index(2);
+    assert_eq!(result, None);
+    assert_eq!(set.len(), 2);
+}
+
+#[test]
+fn sort_unstable_by() {
+    let mut set: IndexSet<i32> = IndexSet::new();
+    set.extend(vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
+    set.sort_unstable_by(|a, b| b.cmp(a));
+    assert_eq!(set.as_slice(), &[10, 9, 8, 7, 6, 5, 4, 3, 2, 1]);
+}
+
+#[test]
+fn sort_by() {
+    let mut set: IndexSet<i32> = IndexSet::new();
+    set.insert(3);
+    set.insert(1);
+    set.insert(2);
+    set.sort_by(|a, b| a.cmp(b));
+    assert_eq!(set.as_slice(), &[1, 2, 3]);
+}
+
+#[test]
+fn drain() {
+    let mut set: IndexSet<i32> = IndexSet::new();
+    set.insert(1);
+    set.insert(2);
+    set.insert(3);
+
+    let drain = set.drain(0..2);
+    assert_eq!(drain.as_slice(), &[1, 2]);
```

Comment:
It would be worth testing the output of the drain iterator, and the state of the set afterward.